File: Lab1/cs_solution/Display.cs
class Display
{
    public int Width;
    public int Height;
    public float Ppi;
    public string Model;

    private int GetArea()
    {
        return Width * Height;
    }

    private void ShowResult(int result, string characteristic, Display m){
        switch (result)
        {
            case -1:
                Console.WriteLine($"{m.Model} has bigger {characteristic} than {Model}");
            break;
            case 1:
                Console.WriteLine($"{Model} has bigger {characteristic} than {m.Model}");
                break;
            case 0:
                Console.WriteLine($"Both monitors have the same {characteristic}");
                break;
            default:
                Console.WriteLine($"Couldn't determine the {characteristic}");
                break;
        }
    }

    public Display(int width, int height, float ppi, string model)
    {
        Width = width;
        Height = height;
        Ppi = ppi;
        Model = model;
    }

    public void CompareSize(Display m)
    {
        int currentSize = GetArea();
        int mSize = m.GetArea();
        int result = currentSize.CompareTo(mSize);

        ShowResult(result, "screen size", m);
    }
    public void CompareSharpness(Display m)
    {
        int result = Ppi.CompareTo(m.Ppi);
        ShowResult(result, "sharpness", m);
    }
    public void CompareWithMonitor(Display m)
    {
        Console.WriteLine($"Overall comparison of Monitors ({Model}, {m.Model})");
        Console.WriteLine("---------------------------------");
        CompareSize(m);
        CompareSharpness(m);
    }
}

File: Lab1/ts_solution/src/modules/Display.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Display = void 0;
var Display = /** @class */ (function () {
    function Display(Width, Height, Ppi, Model) {
        var _this = this;
        this.Width = Width;
        this.Height = Height;
        this.Ppi = Ppi;
        this.Model = Model;
        this.GetArea = function () { return _this.Width * _this.Height; };
    }
    Display.prototype.ShowResult = function (result, characteristic, m) {
        var _a;
        var answers = (_a = {},
            _a[-1] = "".concat(m.Model, " has bigger ").concat(characteristic, " than ").concat(this.Model),
            _a[1] = "".concat(this.Model, " has bigger ").concat(characteristic, " than ").concat(m.Model),
            _a[0] = "Both monitors are of the same size",
            _a);
        console.log(answers[result]);
    };
    Display.prototype.CompareSize = function (m) {
        var characteristic = "screen size";
        var currentSize = this.GetArea();
        var mSize = m.GetArea();
        var diff = currentSize - mSize;
        if (diff === 0)
            this.ShowResult(0, characteristic, m);
        if (diff < 0)
            this.ShowResult(-1, characteristic, m);
        if (diff > 0)
            this.ShowResult(1, characteristic, m);
    };
    Display.prototype.CompareSharpness = function (m) {
        var characteristic = "sharpness";
        var diff = this.Ppi - m.Ppi;
        if (diff === 0)
            this.ShowResult(0, characteristic, m);
        if (diff < 0)
            this.ShowResult(-1, characteristic, m);
        if (diff > 0)
            this.ShowResult(1, characteristic, m);
    };
    Display.prototype.CompareWithMonitor = function (m) {
        console.log("Overall comparison of monitors\n\n                 --------------------------------");
        this.CompareSize(m);
        this.CompareSharpness(m);
    };
    return Display;
}());
exports.Display = Display;
